macros/cross_db_utils/width_bucket.sql

```diff
@@ -0,0 +1,33 @@
+{% macro width_bucket(expr, min_value, max_value, num_buckets) %}
+  {{ adapter_macro('dbt_utils.width_bucket', expr, min_value, max_value, num_buckets) }}
+{% endmacro %}
+
+
+{% macro default__width_bucket(expr, min_value, max_value, num_buckets) -%}
+
+    {% set bucket_width -%}
+    (( {{ max_value }} - {{ min_value }} ) / {{ num_buckets }} )
+    {% endset %}
+    (
+        -- to break ties when the amount is eaxtly at the bucket egde
+        case
+            when mod(
+                        {{ dbt_utils.safe_cast(expr, 'numeric') }}, 
+                        {{ dbt_utils.safe_cast(bucket_width, 'numeric') }}
+                    ) = 0
+            then 1
+            else 0
+        end
+    ) +
+      -- Anything over max_value goes the N+1 bucket
+        least(
+            ceil(
+                ({{ expr }} - {{ min_value }})/{{ bucket_width }}
+            ),
+            {{ num_buckets }} + 1
+        )
+{%- endmacro %}
+
+{% macro snowflake__width_bucket(expr, min_value, max_value, num_buckets) %}
```

Comment:
do we need this macro? I think Snowflake should just be able to use the default implementation, right?

Reply:
I had understood this as: on Snowflake we pass the parameters on to the built-in function. Is that now what you think this is doing?

Reply:
oh!! I so didn't catch that this was how it worked - shout out to Snowflake for having all the good functions :)

Cool, this is perfect, ignore me :)